### DISCUSSION TITLE
CDT task stimuli

### DISCUSSION THREAD

**Discussion by JPR-source** [jspsych/jsPsych#904] [Q&A] [answered]
Hello,

I'm looking to create a Change Detection Task in which participants are shown an array of coloured lines at different angles. I'm at a bit of a loss about how to create the stimuli and how best to implement this: Should I do this on the fly using html/css? Should I generate the stimuli elsewhere and save/present them as images? Would one of these be preferable? Is there a way of creating these grids using Javascript/CSS and pre-loading them so as to avoid lag etc.? Could I create a plug in to create a CDT array?
Any help or information would be much appreciated!

Thanks

**Comment by vijaymarupudi** [comment, marked as answer]
You can use the `html-keyboard-response` plugin for this, and have it render the lines that you'd like in the html. I would personally avoid creating images externally, but that might be easier in some situations.

In html and css, for the lines, you can use an `<hr>` tag. For the rotation, look into the css [`rotate()`].

[`rotate()`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotate

**Comment by JPR-source** [comment]
Thanks, I'll try it that way!
